bin/pasta_show_maintainers.py

```diff
@@ -0,0 +1,299 @@
+                                filter_by_files and filename in filter_by_files)
+                    else:
+                        maintainer_to_subsystem[name] = set([subsystem])
+                        counts[name] = CountItem(filename, loc, byte,
+                                filter_by_files and filename in filter_by_files)
+                elif verbose:
+                    log.info('Ignoring mailing list:'+ str(maintainer))
+    if filter_by_files:
+        # If filter is given, only return the relevant lines:
+        counts = {name:item for name,item in counts.items() 
+                    if item.display_in_filter()}      
+    # Sort maintainers by decreasing total line counts
+    return decreasing_order(counts), maintainer_to_subsystem
+
+def show_maintainers(config, sub, argv):
+    parser = ArgumentParser(description='Display file sizes grouped by ' 
+                                            'maintainers or subsystems')
+    parser.add_argument('show_maintainers', nargs=1, type=bool)
+    parser.add_argument('--entries', action='store_true', 
+                            help='groups by subsystems')
+    parser.add_argument('--verbose', action='store_true', 
+                            help='Show ignored exceptions on the way')
+    parser.add_argument('--smallstat', action='store_true', help='Simple view')
+    parser.add_argument('--bytes', action='store_true', help='Show byte counts')
+    parser.add_argument('--outfile', type=str, help='Output to a csv file')
+    parser.add_argument('--file', type=str, help='Specify MAINTAINERS file')
+    parser.add_argument('--filter', type=str, help='Filter by file list')
+    parser.add_argument('--commit', type=str, 
+                            help='Specify a commit hash for Linux repo')
+    parser.add_argument('--cpu', dest='cpu_factor', metavar='cpu', type=float,
+                        default=1.0, help='CPU factor for parallelisation '
+                                          '(default: %(default)s)')
+    args = parser.parse_args()
+    d_repo = config.repo_location
+    repo = pygit2.Repository(d_repo)
+    commit_version = args.commit if args.commit else 'HEAD'
+    commit = repo.revparse_single(commit_version)
+    tree = commit.tree
+        
+    if args.filter:
+        filter_by_files = file_to_string(args.filter).splitlines()
+        # Automatically find kernel version from file name "v5.6"
+        version_regex = re.compile(r'v[0-9]+.[0-9]+')
+        kernel_version = version_regex.findall(args.filter)[0]
+
+    if args.file:
+        all_maintainers = file_to_string(args.file)
+        LinuxMaintainers(all_maintainers)
+    else:
+        if kernel_version:
+            all_maintainers = load_maintainers(
+                config, [kernel_version]).get(kernel_version)
+        else:
+            all_maintainers = tree['MAINTAINERS'].data.decode('iso8859')
+
+    all_filenames = walk_commit_tree(tree)
```

Comment:
Ok, just a second. Let's get one step back and think of our initial problem:

We want to know the scale/volume of subsystems of the kernel. Either from an overall perspective, or related to a specific configuration. A specific configuration is always mapped to a version and list of affected files. So it is not necessary to specify both: Kernel version and a commit hash for the commit tree walk.

Let me give you two examples:

Get an overall overview of the maintainers of v5.3
$ ./pasta show_maintainers --entries --revision v5.3

Get an overall overview of the maintainers of HEAD
$ ./pasta show_maintainers --entries

Get an overview of the maintainers for a specific configuration, will default to MAINTAINERS of v5.4
$ ./pasta show_maintainers --entries --config defconfig-x86.v5.4.files

Get an overview of the maintainers for a specific configuration, will enforce v5.5, though (probably never required)
$ ./pasta show_maintainers --entries --config defconfig-x86.v5.4.files --revision v5.5